Pop-Grupa-A-Backend/app/main/services/cluster_nodes_service.py

```diff
@@ -1,5 +1,75 @@
 import app.main.repositories.cluster_nodes_repository as nodes_repository
-import app.main.repositories.machines_repository as machine_repository
+import app.main.repositories.machines_repository as machines_repository
+
+from app.main.model.NodeStatus import NodeStatus
+from app.main.model.ClusterNode import ClusterNode
+from app.main.model.Machine import Machine
+
+import random
+from urllib.request import urlopen
 
 def get_nodes_for_user(userId):
-    return nodes_repository.get_cluster_nodes_for_user(userId)
\ No newline at end of file
+    return nodes_repository.get_cluster_nodes_for_user(userId)
+
+def submit_node(node_id):
+    node = nodes_repository.get_node_for_id(node_id)
+    if not node:
+        return {
+                   'status': 'fail',
+                   'message': f"Node with id = {node_id} does not exist",
+               }, 400
+
+    if node.status is NodeStatus.SUBMITTED:
+        return {
+                   'status': 'fail',
+                   'message': f"Node is already submitted.",
+               }, 400
+
+    for machine in node.machines:
+        if not verify_machine(machine):
+            return {
+                       'status': 'fail',
+                       'message': f"Couldn't verify machine with IP = {machine.ip_address}",
+                   }, 400
+
+    node.status = NodeStatus.SUBMITTED
+    nodes_repository.save_changes(node)
+    return {
+               'status': 'success',
+               'message': 'Cluster node successfuly submitted.'
+           }, 201
+
+def create_node(createNodeDto):
```

Comment:
Nie ma sprawdzenia czy user istnieje. Wiem że to powinno byc z sesji wyciągane ale no póki sesji nie ma to trzeba by to zrobić